src/core/ravl_interval.c

```diff
@@ -256,3 +256,44 @@ ravl_interval_data(struct ravl_interval_node *rin)
 {
 	return (void *)rin->addr;
 }
+
+/*
+ * ravl_interval_first -- returns first interval in the tree
+ */
+struct ravl_interval_node *
+ravl_interval_find_first(struct ravl_interval *ri)
+{
+	struct ravl_node *first;
+	first = ravl_first(ri->tree);
+	if (first)
+		return ravl_data(first);
+
+	return NULL;
+}
+
+/*
+ * ravl_interval_first -- returns first interval in the tree
+ */
+struct ravl_interval_node *
+ravl_interval_find_next(struct ravl_interval *ri, void *addr)
+{
+	struct ravl_interval_node range;
```

Comment:
The next element can be retrieved with simple RAVL_PREDICATE_GREATER. So either this function does something else and it's not explained, or the implementation is needlessly complicated.